keel-titus-plugin/src/test/kotlin/com/netflix/spinnaker/keel/titus/verification/ContainerTestVerificationEvaluatorTests.kt

```diff
@@ -0,0 +1,164 @@
+import com.netflix.spinnaker.keel.api.verification.VerificationStatus.FAILED
+import com.netflix.spinnaker.keel.api.verification.VerificationStatus.PASSED
+import com.netflix.spinnaker.keel.api.verification.VerificationStatus.RUNNING
+import com.netflix.spinnaker.keel.orca.ExecutionDetailResponse
+import com.netflix.spinnaker.keel.orca.OrcaExecutionStatus
+import com.netflix.spinnaker.keel.orca.OrcaService
+import com.netflix.spinnaker.keel.test.deliveryConfig
+import com.netflix.spinnaker.keel.titus.batch.RUN_JOB_TYPE
+import de.huxhorn.sulky.ulid.ULID
+import io.mockk.mockk
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.EnumSource
+import strikt.api.expectCatching
+import strikt.api.expectThat
+import strikt.assertions.get
+import strikt.assertions.isEqualTo
+import strikt.assertions.isSuccess
+import java.time.Instant.now
+import io.mockk.coEvery as every
+import io.mockk.coVerify as verify
+
+internal class ContainerTestVerificationEvaluatorTests {
+
+  private val orca = mockk<OrcaService>()
+  private val taskLauncher = mockk<TaskLauncher>()
+  private val subject = ContainerTestVerificationEvaluator(orca, taskLauncher)
+
+  private val context = VerificationContext(
+    deliveryConfig = deliveryConfig(),
+    environmentName = "test",
+    artifactReference = "fnord",
+    version = "1.1"
+  )
+  private val verification = ContainerTestVerification(
+    repository = "illuminati/fnord",
+    location = Location(
+      account = "titustestvpc",
+      region = "ap-south-1"
+    )
+  )
+
+  @Test
+  fun `starting verification launches a container job via orca`() {
+    val taskId = ULID().nextULID()
+
+    every {
+      taskLauncher.submitJob(
+        user = any(),
+        application = any(),
+        notifications = any(),
+        subject = any(),
+        description = any(),
+        correlationId = any(),
+        stages = any()
+      )
+    } answers { Task(id = taskId, name = arg(3)) }
+
+    expectCatching {subject.start(context, verification)}
+      .isSuccess()
+      .get("taskId") isEqualTo taskId
+
+    verify {
+      taskLauncher.submitJob(
+        user = any(),
+        application = any(),
+        notifications = any(),
+        subject = any(),
+        description = any(),
+        correlationId = any(),
+        stages = match {
+          it.first()["type"] == RUN_JOB_TYPE
+        }
+      )
+    }
+  }
+
+  @ParameterizedTest(name = "verification is considered still running if orca status is {0}")
+  @EnumSource(
+    OrcaExecutionStatus::class,
+    names = ["NOT_STARTED", "RUNNING", "PAUSED", "SUSPENDED", "BUFFERED"]
+  )
+  fun `verification is considered still running if orca task is running`(taskStatus: OrcaExecutionStatus) {
+    val taskId = ULID().nextULID()
+    val previousState = runningState(taskId)
+
+    every {
+      orca.getOrchestrationExecution(taskId, any())
```

Comment:
Side-comment: this just made me realize that we're leaking from the `TaskLauncher` abstraction. You can launch jobs using the abstract type, but not query for status...

Reply:
Yes. I think this requires some thought in terms of handling the response coming back from Orca. There's limited value if we just pass it thru. If we can marshal it into a useful format that exposes error information in a consistent way it will make a lot of sense. I think to do it properly is outside of the scope of this PR tho. Maybe in the next one which will involve surfacing information about test results to the user.